citesphere/src/main/webapp/WEB-INF/views/admin/apps/show.jsp

```diff
@@ -17,21 +18,88 @@ $(function() {
         }
     });
 });
+
+$(function() {
+	$(".delete-link").click(function(event) {
+		var id = $(this).data('client-id');
+		$("#delete-client-button").attr("data-client-id", id);
+		$("#confirm-client-entry-id").text(id);
+		$('#client-delete-confirmation').modal("show");	
+		event.preventDefault();
+	});
+	$("#delete-client-button").click(function() {
+		var id = $(this).data("client-id");
+		$.ajax({
+			'url': '<c:url value="/admin/apps/" />' + id + "?${_csrf.parameterName}=${_csrf.token}",
+			'type': "DELETE",
+			'success': function(data) {
+				$("#tr-" + id).remove();
+				$.notify('<i class="fas fa-check-circle"></i> App successfully deleted!', {
+					type: 'success',
+					offset: {
+						x: 50,
+						y: 90
+					},
+					animate: {
+						enter: 'animated fadeInRight',
+						exit: 'animated fadeOutRight'
+					}
+				});
+			},
+			'error': function(data) {
+				$.notify('<i class="fas fa-exclamation-circle"></i> App could not be deleted!', {
+					type: 'danger',
+					offset: {
+						x: 50,
+						y: 90
+					},
+					animate: {
+						enter: 'animated fadeInRight',
+						exit: 'animated fadeOutRight'
+					}
+				});
+			}
+		});
+		$('#client-delete-confirmation').modal("hide");
+		event.preventDefault();
+		$(this).removeData();
```

Comment:
does this mean you remove the table row even if the app could not be deleted?

Reply:
I added this because once row was removed, its clientId was somehow cached and when I was trying to delete some other row It was trying to delete the client with cached clientId.

Reply:
Let's find out what the actual problem there is. Because if you always delete the data no matter if deletion actually succeeded or not, this will have side effects afterwards I would think.

Reply:
I tried to check by explicitly generating error while deleting 2 apps back to back. In case of error table row is not getting removed. removeData() method allows us to remove values that were previously set using jQuery.data()

Reply:
Ah, I see, it's on the modal button. That makes sense.